File: ios/Kurir/modules/auth/signup/SignupModule.swift
//
//  SignupModule.swift
//  Kurir
//
//  Created by Loren on 1/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class SignupModule: Module {
    override func inject() {
        // Wrapper
        factory(SignupWrapper.self) { r in
            SignupWrapper(repository: r.resolve(UsersRepository.self)!)
        }
        
        // Actor
        factory(SignupActor.self) { (r, scene: SignupScene) -> SignupActor in
            SignupActor(scene: scene, wrapper: r.resolve(SignupWrapper.self)!)
        }
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/signup/SignupModule.kt
package au.com.crazybean.mobilex.kurir.modules.auth.signup

import au.com.crazybean.mobilex.kurir.extension.qualifier
import au.com.crazybean.mobilex.kurir.modules.base.ViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val signupModule = module {
    val qualifier = qualifier<SignupWrapper>()

    // Wrapper via ViewModel
    viewModel(qualifier) {
        ViewModel(SignupWrapper(get()))
    }

    // Actor
    factory { (scene: SignupScene?) ->
        val viewModel = get<ViewModel<SignupWrapper>>(qualifier)
        SignupActor(scene, viewModel.wrapper)
    }
}